### ISSUE TITLE
Documentation should note that GEM_HOST_API_KEY env var only works with rubygems >= 3.0.5

### ISSUE THREAD

**Issue by dchandekstark** [rubygems/rubygems#4607]
Support for using GEM_HOST_API_KEY with `gem push` was added in https://github.com/rubygems/rubygems/commit/42c1b42c870e10651a134e90235b2eb74eec6933.

As of this date, the official Ruby 2.6 Docker image, e.g., has rubygems 3.0.3.1, which does not support the env var.

**Comment by deivid-rodriguez** [comment]
Hi @dchandekstark!

`gem push --help` of version `3.0.3.1` shouldn't document this at all, and the online command reference at https://guides.rubygems.org/command-reference/ is always generated from the CLI help of the latest version. Do you have any suggestions on how to clarify this? Bundler has "per-version" documentation at https://bundler.io/docs.html, where the version you're browsing documentation for is more apparent, but we do not have such feature for rubygems documentation.

**Comment by dchandekstark** [comment]
Hi @deivid-rodriguez.  Thanks for the clarification.  I did not compare with the CLI help, but I see now that the web documentation states that it was generated.  Since I am not familiar with how this project is managed, I'm not sure what to suggest. If it helps, a contributing factor to my confusion was that my development environment with Ruby 2.6 has rubygems 3.0.8:

```
$ gem --version
3.0.8
$ ruby --version
ruby 2.6.6p146 (2020-03-31 revision 67876) [x86_64-darwin19]
```

So I was able to use the env var and then assumed that it would work with a close Ruby 2.6 version in a different environment, so was confused to discover it did not.  

```
$ docker run --rm ruby:2.6 gem --version
3.0.3.1
$ docker run --rm ruby:2.6 ruby --version
ruby 2.6.7p197 (2021-04-05 revision 67941) [x86_64-linux]
```

Of course I was able to work around the issue ultimately by running `gem update --system` in the container.  I don't recall updating rubygems in the RVM-installed Ruby on my system, so I can't account for why an earlier version of Ruby would have a later version of rubygems. (Unfortunately it seems I can't build Ruby 2.6.7 b/c https://bugs.ruby-lang.org/issues/17777).

In summary, perhaps this issue itself (as closed) can serve as a sort of documentation for anyone who might encounter this problem?

**Comment by deivid-rodriguez** [comment]
I think `RVM` updates rubygems by default, but docker images don't do that. I think that's why. I think this is fine to close, but it might still help someone running into something similar, indeed. Thanks for reporting it! :+1:
